File: et-ejb/src/main/java/ru/vmakarenko/services/ExpensesService.java
package ru.vmakarenko.services;

import ru.vmakarenko.entities.Expense;
import ru.vmakarenko.entities.ExpensesFilter;
import ru.vmakarenko.entities.User;

import java.util.List;

/**
 * Created by VMakarenko on 2/4/15.
 */
public interface ExpensesService {

    Expense getById(Long id);

    List<Expense> getAll(ExpensesFilter filter);

    Expense update(Expense expense);

    Expense create(Expense expense, User user);

    void delete(Long id);
}

File: fes-web/src/main/webapp/assets/js/expenses/services/expenses-service.js
/**
 * Created by VMakarenko on 2/7/15.
 */

angular.module("app")
    .service("ExpensesService", ['$http',
function ExtInvestigationRequestsService($http ) {
    var url ="api/expenses";

    this.getList = function(filter){
        return $http.post(url + "/all", filter);
    };

    this.getById = function(id){
        return $http.get(url + "/" + id);
    };

    this.createExpense = function(expense){
        return $http.post(url, expense);
    };

    this.updateExpense = function(expense){
        return $http.put(url, expense);
    };

    this.deleteExpense = function(id){
        return $http.delete(url + "/" + id);
    };

    }
]);

